remill/Arch/Instruction.h

```diff
@@ -36,7 +36,8 @@ class Operand {
     kTypeRegister,
     kTypeShiftRegister,
     kTypeImmediate,
-    kTypeAddress
+    kTypeAddress,
+    kTypePointer
```

Comment:
What is the value-added of a `kTypePointer`? It seems like we can use either a `kTypeImmediate` or `kTypeAddress`. Really, though, introducing a new type means updating McSema :-P

Reply:
It's similar (though not identical) to a unsigned `kTypeImmediate`:  
- `InstructionLifter::LiftPointerOperand` is mostly identical to `InstructionLifter::LiftImmediateOperand`), but
- `Arch.cpp::DecodePointer` is different from `Arch.cpp::DecodeImmediate`

I still don't know whether merging them is better or not.

Reply:
Well what I mean is that DecodePointer can produce an immediate.

Reply:
Yes, it's possible.
Do you want me to modify the PR?

Reply:
Yes.

Reply:
I will do this tomorrow because it's late now (in France).

Reply:
À demain.

Reply:
Merci, à demain.

Reply:
@pgoodman I've changed the PR to remove `kTypePointer` and use `kTypeImmediate` instead.